### PR TITLE
Merged changes for apply on forward into api v2.

### PR DESCRIPTION
**Pull request by song-jiang**
## Description
Added changes for apply on forward to api v2.

### CODE REVIEW

File: lib/backend/k8s/resources/globalnetworkpolicies_test.go [hunk 1]
```diff
@@ -116,19 +116,19 @@ var _ = Describe("Global Network Policies conversion methods", func() {
...
 			res1.Spec.PreDNAT = true
-			//TODO: Add back
-			//res1.Spec.ApplyOnForward = true
+			res1.Spec.ApplyOnForward = true
 		})
 
 		AfterEach(func() {
 			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.DoNotTrack = true
 			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.PreDNAT = false
+			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.ApplyOnForward = true
```
Comment (fasaxc) [inline lib/backend/k8s/resources/globalnetworkpolicies_test.go]: Not sure I grok what this is doing but should this reverse the change made in `BeforeEach`?

File: lib/backend/k8s/resources/globalnetworkpolicies_test.go [hunk 2]
```diff
@@ -116,19 +116,19 @@ var _ = Describe("Global Network Policies conversion methods", func() {
 		BeforeEach(func() {
 			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.DoNotTrack = false
 			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.PreDNAT = true
+			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.ApplyOnForward = true
 			res1.Spec.DoNotTrack = false
 			res1.Spec.PreDNAT = true
-			//TODO: Add back
```
Comment (bcreane) [inline lib/backend/k8s/resources/globalnetworkpolicies_test.go]: looks like this comment can go

File: lib/backend/k8s/resources/globalnetworkpolicies_test.go [hunk 3]
```diff
@@ -116,19 +116,19 @@ var _ = Describe("Global Network Policies conversion methods", func() {
...
+			res1.Spec.ApplyOnForward = true
 		})
 
 		AfterEach(func() {
 			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.DoNotTrack = true
 			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.PreDNAT = false
+			kvp1.Value.(*apiv2.GlobalNetworkPolicy).Spec.ApplyOnForward = true
 			res1.Spec.DoNotTrack = true
 			res1.Spec.PreDNAT = false
-			//TODO: Add back
```
Comment (bcreane) [inline lib/backend/k8s/resources/globalnetworkpolicies_test.go]: and this

File: lib/backend/syncersv1/updateprocessors/networkpolicyprocessor.go [hunk 4]
```diff
@@ -65,6 +65,7 @@ func convertPolicyV2ToV1Spec(spec apiv2.PolicySpec) (interface{}, error) {
 		Selector:      spec.Selector,
 		DoNotTrack:    spec.DoNotTrack,
 		PreDNAT:       spec.PreDNAT,
+		ApplyOnForward: spec.ApplyOnForward,
```
Comment (bcreane) [inline lib/backend/syncersv1/updateprocessors/networkpolicyprocessor.go]: indent looks off, may run `go fmt`?

Comment (song-jiang) [reply lib/backend/syncersv1/updateprocessors/networkpolicyprocessor.go]: Strange, my editor shows correct indent. Will fix that.

File: lib/backend/k8s/resources/globalnetworkpolicies_test.go (hunk 1 above)
Comment (song-jiang) [reply lib/backend/k8s/resources/globalnetworkpolicies_test.go]: ApplyOnForward is always True if either doNotTrack or PreDNAT is true.

File: lib/backend/k8s/resources/globalnetworkpolicies_test.go (hunk 3 above)
Comment (song-jiang) [reply lib/backend/k8s/resources/globalnetworkpolicies_test.go]: Removed.

File: lib/backend/k8s/resources/globalnetworkpolicies_test.go (hunk 2 above)
Comment (song-jiang) [reply lib/backend/k8s/resources/globalnetworkpolicies_test.go]: Removed.

**Pull request merged**